src/Microsoft.Framework.DependencyInjection/ServiceLookup/ServiceScope.cs

```diff
@@ -19,9 +21,19 @@ public IServiceProvider ServiceProvider
             get { return _scopedProvider; }
         }
 
+        internal void Reset()
```

Comment:
Why internal here? It would also be helpful to capture your explanation of this in comments.